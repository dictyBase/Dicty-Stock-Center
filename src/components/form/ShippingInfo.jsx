import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { Flex, Box } from 'rebass'
import {
  RequiredText,
  FormGroup,
  FormControl,
  HelpBlock,
  ControlLabel,
  RadioInline
} from 'styles'

export default class ShippingInfo extends Component {
  displayName = 'form shipping information'

  static propTypes = {
      shipAccount: PropTypes.object.isRequired,
      shipAccountNum: PropTypes.object.isRequired
  }

  renderShipAccountNum = () => {
      const { shipAccountNum } = this.props
      return (
      <FormGroup>
        <Flex wrap justify="center">
          <Box w={ 1 / 5 } />
          <Box w={ ['90%', '90%', 2 / 3] }>
            <FormControl
              {...shipAccountNum}
              placeholder="Shipping Account Number"
            />
            { shipAccountNum.error && (
              <HelpBlock>{ shipAccountNum.error }</HelpBlock>
            ) }
          </Box>
        </Flex>
      </FormGroup>
    )
  }

  render() {
      const { shipAccount } = this.props
      const hasError = shipAccount.touched && shipAccount.error
      return (
      <FormGroup>
        <Flex wrap justify="center">
          <Box w={ 1 / 4 } ml={ 1 } mb={ 2 }>
            <ControlLabel>
              <RequiredText>* </RequiredText>
              Shipping Account:
            </ControlLabel>
          </Box>
          <Box w={ 2 / 3 } mr={ 2 } mb={ 1 }>
            <RadioInline>
              <input
                type="radio"
                {...shipAccount}
                value="Fedex"
                checked={ shipAccount.value === 'Fedex' }
              />&nbsp; FedEx
            </RadioInline>
            <RadioInline>
              <input
                type="radio"
                {...shipAccount}
                value="UPS"
                checked={ shipAccount.value === 'UPS' }
              />&nbsp; UPS
            </RadioInline>
            <RadioInline>
              <input
                type="radio"
                {...shipAccount}
                value="DHL"
                checked={ shipAccount.value === 'DHL' }
              />&nbsp; DHL
            </RadioInline>
            <RadioInline>
              <input
                type="radio"
                {...shipAccount}
                value="Will call 1-312-503-4169"
                checked={ shipAccount.value === 'Will call 1-312-503-4169' }
              />&nbsp; Call in Credit card # for FedEx Waybill 1-312-503-4169
            </RadioInline>
            { hasError && <HelpBlock>{ shipAccount.error }</HelpBlock> }
          </Box>
        </Flex>
        { !(shipAccount.value === 'Will call 1-312-503-4169') &&
          this.renderShipAccountNum() }
      </FormGroup>
    )
  }
}
