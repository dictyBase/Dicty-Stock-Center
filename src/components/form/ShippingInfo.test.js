import React from "react"
import renderer from "react-test-renderer"
import "jest-styled-components"
import ShippingInfo from "./ShippingInfo"

test("matching a snapshot of ShippingInfo", () => {
  const shipAccount = {
    value: "Fedex",
    touched: true
  }
  const shipAccountNum = {
    value: "",
    touched: false
  }

  const component = renderer.create(
    <ShippingInfo shipAccount={shipAccount} shipAccountNum={shipAccountNum} />
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
