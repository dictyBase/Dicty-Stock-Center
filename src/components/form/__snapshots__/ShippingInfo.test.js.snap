// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matching a snapshot of ShippingInfo 1`] = `
.c2 {
  color: #dc3545;
}

.c1 {
  display: inline-block;
  padding-top: 7px;
  margin-bottom: 0;
  text-align: right;
  font-weight: 600;
}

.c0 {
  margin-right: -15px;
  margin-left: -15px;
}

.c4 {
  display: block;
  width: 100%;
  height: 34px;
  padding: 6px 12px;
  font-size: 14px;
  line-height: 1.42857143;
  color: #555;
  background-color: #fff;
  background-image: none;
  border: 1px solid #ccc;
  border-radius: 4px;
  -webkit-box-shadow: inset 0 1px 1px rgba(0,0,0,0.075);
  box-shadow: inset 0 1px 1px rgba(0,0,0,0.075);
  -webkit-transition: border-color ease-in-out 0.15s, -webkit-box-shadow ease-in-out 0.15s;
  -o-transition: border-color ease-in-out 0.15s,box-shadow ease-in-out 0.15s;
  -webkit-transition: border-color ease-in-out 0.15s,box-shadow ease-in-out 0.15s;
  transition: border-color ease-in-out 0.15s,box-shadow ease-in-out 0.15s;
}

.c4:focus {
  border-color: #66afe9;
  outline: 0;
  -webkit-box-shadow: inset 0 1px 1px rgba(0,0,0,0.075), 0 0 8px rgba(102,175,233,0.6);
  box-shadow: inset 0 1px 1px rgba(0,0,0,0.075), 0 0 8px rgba(102,175,233,0.6);
}

.c3 {
  position: relative;
  display: inline-block;
  padding-right: 20px;
  margin-bottom: 0;
  font-weight: normal;
  vertical-align: middle;
  cursor: pointer;
}

<div
  className="c0"
>
  <div
    className="sc-htpNat eDtRyC sc-bdVaJa iHZvIS"
    is={null}
  >
    <div
      className="cyncUw"
      is={null}
    >
      <label
        className="c1"
      >
        <span
          className="c2"
        >
          * 
        </span>
        Shipping Account:
      </label>
    </div>
    <div
      className="XupaC"
      is={null}
    >
      <label
        className="c3"
      >
        <input
          checked={true}
          touched={true}
          type="radio"
          value="Fedex"
        />
          FedEx
      </label>
      <label
        className="c3"
      >
        <input
          checked={false}
          touched={true}
          type="radio"
          value="UPS"
        />
          UPS
      </label>
      <label
        className="c3"
      >
        <input
          checked={false}
          touched={true}
          type="radio"
          value="DHL"
        />
          DHL
      </label>
      <label
        className="c3"
      >
        <input
          checked={false}
          touched={true}
          type="radio"
          value="Will call 1-312-503-4169"
        />
          Call in Credit card # for FedEx Waybill 1-312-503-4169
      </label>
    </div>
  </div>
  <div
    className="c0"
  >
    <div
      className="sc-htpNat eDtRyC sc-bdVaJa iHZvIS"
      is={null}
    >
      <div
        className="cyncUw"
        is={null}
      />
      <div
        className="cjDWjg"
        is={null}
      >
        <input
          className="c4"
          placeholder="Shipping Account Number"
          value=""
        />
      </div>
    </div>
  </div>
</div>
`;
